### PR TITLE
Add functions for getting...

### PR DESCRIPTION
**Pull request by ostowe**
- Category (one term)
- Custom tax (one term)

### CODE REVIEW

File: inc/wp-components/classes/integrations/class-google-tag-manager.php [hunk 1]
```diff
@@ -94,31 +94,54 @@ public function get_post_id() : string {
...
 	 *
+	 * @param  string $taxonomy Taxonomy for which terms should be extracted.
 	 * @return array
 	 */
-	public function get_tags() : array {
+	public function get_taxonomy_terms( $taxonomy ) : array {
 		// Single article.
 		if ( $this->query->is_single() ) {
-			$tags = wp_get_post_tags( $this->query->post->ID ?? 0 );
+			$terms = wp_get_post_terms( $this->query->post->ID, $taxonomy );
```
Comment (amyevans) [inline inc/wp-components/classes/integrations/class-google-tag-manager.php]: If the taxonomy doesn't exist, `wp_get_post_terms()` returns a `\WP_Error`. It couldn't hurt to check for that and return an empty array early before doing the `array_map`

File: inc/wp-components/classes/integrations/class-google-tag-manager.php [hunk 2]
```diff
@@ -94,31 +94,54 @@ public function get_post_id() : string {
...
+	 */
+	public function get_single_taxonomy_term( $taxonomy ) {
+		// Single article.
+		if ( $this->query->is_single() ) {
+			$terms = wp_get_post_terms( $this->query->post->ID, $taxonomy );
+
+			return $terms[0]->name ?? null;
+		}
+
+		// Section taxonomy landing.
```
Comment (amyevans) [inline inc/wp-components/classes/integrations/class-google-tag-manager.php]:
```suggestion
		// Taxonomy landing.
```

File: inc/wp-components/classes/integrations/class-google-tag-manager.php [hunk 3]
```diff
@@ -94,31 +94,54 @@ public function get_post_id() : string {
...
+	/**
+	 * Get taxonomy term, if applicable.
+	 *
+	 * @param  string $taxonomy Taxonomy for which the first term should be extracted.
+	 * @return string|null
+	 */
+	public function get_single_taxonomy_term( $taxonomy ) {
+		// Single article.
+		if ( $this->query->is_single() ) {
+			$terms = wp_get_post_terms( $this->query->post->ID, $taxonomy );
```
Comment (amyevans) [inline inc/wp-components/classes/integrations/class-google-tag-manager.php]: Same comment wrt checking for a WP_Error first

File: inc/wp-components/classes/integrations/class-google-tag-manager.php (hunk 1 above)
Comment (ostowe) [reply inc/wp-components/classes/integrations/class-google-tag-manager.php]: Cood call, will do

**Pull request merged**